webpack.config.js

```diff
@@ -379,6 +380,112 @@ var config = {
     ],
 };
 
+async function initializeModuleFederation() {
+    function makeSingletonSharedModules(packageNames) {
+        const sharedObject = {};
+
+        for (const packageName of packageNames) {
+            const version = packageJson.dependencies[packageName];
+
+            sharedObject[packageName] = {
+                requiredVersion: version,
+                singleton: true,
+                version,
+            };
+        }
+
+        return sharedObject;
+    }
+
+    function isWebpackDevServerAvailable(baseUrl) {
+        return new Promise((resolve) => {
+            if (!DEV) {
+                resolve(false);
+                return;
+            }
+
+            const req = http.request(`${baseUrl}/remote_entry.js`, (response) => {
+                return resolve(response.statusCode === 200);
+            });
+
+            req.on('error', () => {
+                resolve(false);
+            });
+
+            req.end();
+        });
+    }
+
+    async function getRemoteModules() {
+        const products = [
+            {name: 'focalboard', baseUrl: 'http://localhost:9006'},
+        ];
+
+        const productsFound = await Promise.all(products.map((product) => isWebpackDevServerAvailable(product.baseUrl)));
+
+        const remotes = {};
+        const aliases = {};
+
+        for (let i = 0; i < products.length; i++) {
+            const product = products[i];
+            const found = productsFound[i];
+
+            if (found) {
+                console.log(`Product ${product.name} found, adding as remote module`);
+
+                remotes[product.name] = `${product.name}@${product.baseUrl}/remote_entry.js`;
+            } else {
+                console.log(`Product ${product.name} not found`);
+
+                // Add false aliases to prevent Webpack from trying to resolve the missing modules
+                aliases[product.name] = false;
+                aliases[`${product.name}/manifest`] = false;
```

Comment:
As the comment above says, we have to tell Webpack not to even try to load this module. Without this, the web app will still know not to load the product (because the `REMOTE_MODULES` constant won't include its name), but Webpack will still see the `import('focalboard')` and print an error that it can't find it